File: frontend/src/types/Order.ts
import type { MenuItemSimple, MenuItemSummary } from "./Menu";

export type OrderItemCreate = {
    menuItem: number;
    menuItemSummary: MenuItemSummary;
    half_tray_quantity: number;
    full_tray_quantity: number;
}
export type OrderCreate = {
    orderItems: OrderItemCreate[];
    customer_name: string;
    customer_phone_number: string;
    customer_email: string;
    delivery_time: Date;
    price: number;
}
export type OrderItem = {
    id: number;
    menuItem: MenuItemSimple;
    half_tray_quantity: number;
    full_tray_quantity: number;
}
export type Order = {
    id: number;
    orderItems: OrderItem[];
    customer_name: string;
    customer_phone: string;
    customer_email: string;
    delivery_time: string;
    price: number;
    created_at: string;
}

File: backend/src/main/java/com/backend/order/Order.java
package com.backend.order;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import org.springframework.web.bind.annotation.Mapping;

import java.time.ZonedDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name= "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<OrderItem> orderItems = new ArrayList<>();
    private Double price;

    // Customer details
    private String customer_name;
    private String customer_phone_number;
    private String customer_email;

    private final ZonedDateTime created_at = ZonedDateTime.now();

    // Delivery details
    private ZonedDateTime delivery_time;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public List<OrderItem> getOrderItems() {
        return orderItems;
    }

    public void setOrderItems(List<OrderItem> orderItems) {
        this.orderItems = orderItems;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public String getCustomer_name() {
        return customer_name;
    }

    public void setCustomer_name(String customer_name) {
        this.customer_name = customer_name;
    }

    public String getCustomer_phone_number() {
        return customer_phone_number;
    }

    public void setCustomer_phone_number(String customer_phone_number) {
        this.customer_phone_number = customer_phone_number;
    }

    public String getCustomer_email() {
        return customer_email;
    }

    public void setCustomer_email(String customer_email) {
        this.customer_email = customer_email;
    }

    public ZonedDateTime getDelivery_time() {
        return delivery_time;
    }

    public ZonedDateTime getCreated_at() {
        return created_at;
    }

    public void setDelivery_time(ZonedDateTime delivery_time) {
        this.delivery_time = delivery_time;
    }
}
